contracts/HybridCustody.cdc

```diff
@@ -124,97 +133,96 @@ pub contract HybridCustody {
 
         /// Returns the current owner of this account, if there is one
-        pub fun getOwner(): Address?
+        access(all) view fun getOwner(): Address?
 
         /// Returns the pending owner of this account, if there is one
-        pub fun getPendingOwner(): Address?
+        access(all) view fun getPendingOwner(): Address?
 
         /// A callback which is invoked when a parent redeems an owned account
         access(contract) fun setOwnerCallback(_ addr: Address)
         
         /// Destroys all outstanding AuthAccount capabilities on this owned account, and creates a new one for the
         /// OwnedAccount to use
-        pub fun rotateAuthAccount()
+        access(Owner) fun rotateAuthAccount()
 
         /// Revokes all keys on this account
-        pub fun revokeAllKeys()
+        access(Owner) fun revokeAllKeys()
     }
 
     /// Public methods exposed on a ChildAccount resource. OwnedAccountPublic will share some methods here, but isn't
     /// necessarily the same.
     ///
-    pub resource interface AccountPublic {
-        pub fun getPublicCapability(path: PublicPath, type: Type): Capability?
-        pub fun getPublicCapFromDelegator(type: Type): Capability?
-        pub fun getAddress(): Address
-        pub fun getCapabilityFactoryManager(): &{CapabilityFactory.Getter}?
-        pub fun getCapabilityFilter(): &{CapabilityFilter.Filter}?
+    access(all) resource interface AccountPublic {
+        access(all) view fun getPublicCapability(path: PublicPath, type: Type): Capability?
+        access(all) view fun getPublicCapFromDelegator(type: Type): Capability?
+        access(all) view fun getAddress(): Address
+        access(all) view fun getCapabilityFactoryManager(): &{CapabilityFactory.Getter}?
+        access(all) view fun getCapabilityFilter(): &{CapabilityFilter.Filter}?
+        access(all) view fun getControllerIDForType(type: Type, forPath: StoragePath): UInt64?
     }
 
     /// Methods accessible to the designated parent of a ChildAccount
     ///
-    pub resource interface AccountPrivate {
-        pub fun getCapability(path: CapabilityPath, type: Type): Capability? {
+    access(all) resource interface AccountPrivate {
+        access(Child) view fun getCapability(controllerID: UInt64, type: Type): Capability? {
             post {
                 result == nil || [true, nil].contains(self.getManagerCapabilityFilter()?.allowed(cap: result!)):
                     "Capability is not allowed by this account's Parent"
             }
         }
-        pub fun getPublicCapability(path: PublicPath, type: Type): Capability?
-        pub fun getManagerCapabilityFilter():  &{CapabilityFilter.Filter}?
-        pub fun getPublicCapFromDelegator(type: Type): Capability?
-        pub fun getPrivateCapFromDelegator(type: Type): Capability? {
+        access(all) view fun getManagerCapabilityFilter():  &{CapabilityFilter.Filter}?
```

Comment:
Should this be `access(Child)`?

Reply:
So `CapabilityFilter` is intentionally public. One such use for making this a public method would be for someone to borrow a particular filter and make sure certain provider capabilities are valid or not